### PR TITLE
1.17.0 release notes

### PR DESCRIPTION
**Pull request by salahalsaleh**

### CODE REVIEW

File: content/release-notes/1.17.0.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+date: 2020-07-13T08:00:00-07:00
+linktitle: "1.17.0"
+title: "1.17.0"
+kubernetes: "1.16, 1.17, and 1.18"
+---
+
+{{<features>}}
+* GitOps - Added Github Enterprise support.
+* Snapshots - Added a "Internal Storage (Default)" option to the providers list for KURL embedded clusters. Which will switch back to the default configuration that velero was set up with when the app was installed.
```
Comment (salahalsaleh) [inline content/release-notes/1.17.0.md:10]: @austinchambers there's probably a better wording for this. any ideas?

File: content/release-notes/1.17.0.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+---
+date: 2020-07-13T08:00:00-07:00
+linktitle: "1.17.0"
+title: "1.17.0"
+kubernetes: "1.16, 1.17, and 1.18"
+---
+
+{{<features>}}
+* GitOps - Added Github Enterprise support.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:9]: This is a bug, not a new feature. That being said, I understand why we want to have this listed as a "new feature." How about we treat the bug as "We incorrectly allowed GitHub Enterprise as an option when it wasn't supported yet." Something like that. 

```suggestion
* Admin Console now supports GitHub Enterprise as a GitOps repository. 
```

File: content/release-notes/1.17.0.md (hunk 1 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:10]:
```suggestion
* Ability to change to a different snapshot storage provider using the UI on the Snapshot Settings page. 
* "Internal Storage" snapshot provider option has been added to the snapshot providers list for kURL embedded clusters. This allows the user to easily return to the default snapshot configuration when another provider (such as AWS S3 or Google Storage) has been configured. 
```

File: content/release-notes/1.17.0.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* Snapshots - Added a "Internal Storage (Default)" option to the providers list for KURL embedded clusters. Which will switch back to the default configuration that velero was set up with when the app was installed.
+* Added `when` field support for config groups.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:11]:
```suggestion
* Added `when` field support for config groups [https://kots.io/reference/v1beta1/config/#when](https://kots.io/reference/v1beta1/config/#when)
```

Comment (austinchambers) [reply content/release-notes/1.17.0.md:11]: Verify the above syntax renders properly. I think I got it right.

File: content/release-notes/1.17.0.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{{</features>}}
+
+{{<changes>}}
+* Kotsadm - New Support bundles will be shown at the top of the list instead of the bottom.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:15]:
```suggestion
* The most recent support bundles on the Admin Console troubleshoot page are now shown first. 
```

File: content/release-notes/1.17.0.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+* Kotsadm - Showing warnings on the Version History page in the Admin Console when installing an application with invalid YAML.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:16]:
```suggestion
* Helpful warnings now appear on the Version History page if the user attempts to deploy a version of the application with invalid YAML. 
```

File: content/release-notes/1.17.0.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+* Kotsadm - Added a page title for the 'View files' and 'Config' tabs.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:17]: This is kind of small. I'd remove. 
```suggestion
```

File: content/release-notes/1.17.0.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+* Removed the KOTS metadata that was added to all application manifests. Which included the local application and the version sequence injected as annotations to all deployed manifests.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:18]: Why did we do this?

File: content/release-notes/1.17.0.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+* Show the airgap upload progress percentage more granularly.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:19]:
```suggestion
* Improved the granularity of the airgap upload progress indicator.
```

File: content/release-notes/1.17.0.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+* Removed the "Re-analyze" button from the troubleshoot support bundle analysis page.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:20]: Why was this removed?

File: content/release-notes/1.17.0.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+* Auto deploy the app's first version when installing the app and no preflight checks are available or preflight checks pass.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:21]: I thought we already did this? Or is this just for existing clusters? What was the experience like when we didn't do this and no preflights were configured?

File: content/release-notes/1.17.0.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+* Update to import k8s 1.18 libraries.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:22]: 🤷  Support for later versions of k8s is one thing, but I don't care about which libraries are imported. 
```suggestion
```

File: content/release-notes/1.17.0.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+* Show a descriptive error message when `kubectl kots download` is called with an app slug that does not exist.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:23]:
```suggestion
* Improved error message when `kubectl kots download` is called with an app slug that does not exist.
```

File: content/release-notes/1.17.0.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+{{</changes>}}
+
+{{<fixes>}}
+* Fixed a bug in KOTS CLI where using `--deploy` flag when checking for updates does not deploy the latest release.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:27]:
```suggestion
* Fixed a bug in KOTS CLI where the `--deploy` flag when checking for [upstream](https://kots.io/kots-cli/upstream/) updates fails to deploy the latest release.
```

File: content/release-notes/1.17.0.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed a bug in KOTS CLI where automated installs fail to deploy if no preflight checks are available.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:28]: Which CLI command is this for? Guessing 'install'? 

```suggestion
* Fixed a bug in KOTS CLI [install command](https://kots.io/kots-cli/install/) where automated installs fail to deploy if no preflight checks are configured for the application. 
```

File: content/release-notes/1.17.0.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed a bug in KOTS CLI where checking for updates shows `There are no application updates available` even when updates are actually available.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:29]: Which command was this for? Is it `kots version`? Is it `upstream`?

File: content/release-notes/1.17.0.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed an issue where license expiration isn't checked until after fetching application.
+* Fixed a bug where preflight checks were not triggered for airgap or automated installs.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:31]: Does this always happen, or just in a specific situation? If so, what situation? If not, I'd use stronger wording to convey this is seriously broken (and probably put this at the top of the bug fixes list). Also, what do you mean by `automated installs`? Is this an install via `kots install?`

File: content/release-notes/1.17.0.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed a bug where Snapshots scheduled `@daily` run just on day 1 of the month instead of every day.
+* Fixed an issue in Kotsadm troubleshoot "View Files" tab that shows full path instead of just subdirectory.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:33]:
```suggestion
* Fixed an issue on the "View Files" tab in the Admin Console that shows full path instead of just subdirectory.
```

File: content/release-notes/1.17.0.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed an issue where the minio image included a binary with CAP_CHOWN set.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:34]: Ok... but what's the issue? What happens when this binary is included? If there's no user impact, I'd probably remove this line.

File: content/release-notes/1.17.0.md [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed button styles on TLS page on KOTS cluster installs.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:35]: This was a visual glitch. 

```suggestion
* Fixed visual glitches on the buttons of the [TLS certificate page](https://kurl.sh/docs/install-with-kurl/setup-tls-certs) that appears during embedded cluster installation. 
```

File: content/release-notes/1.17.0.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed snapshot hooks status.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:36]: Not sure what this is. Need more detail, I think. How did this bug manifest to the end user?

File: content/release-notes/1.17.0.md [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed backup size on snapshots list.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:37]: Is this when the backup size incorrectly says `0 bytes` or `undefined` or something like that? 

If so, I'd explicitly state that.

File: content/release-notes/1.17.0.md [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed Velero checks on Snapshot settings.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:38]: Not sure what this is about. Is this about the checks that fail because we require elevated permissions? What's the user experience here?

File: content/release-notes/1.17.0.md [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed a bug where clicking `Save changes` in the "Registry settings" tab without changing anything breaks the registry.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:39]: I'd put this near the top of the bug list. Corruption is high severity. 

```suggestion
* Fixed a bug where clicking `Save changes` in the "Registry settings" tab without changing anything permanently breaks the registry connection to the Admin Console. Repairing this connection then requires manual intervention outside the Admin Console. 
```

File: content/release-notes/1.17.0.md [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed an issue where the diff viewer in Kotsadm does not scroll to the end of the file.
+* Fixed an issue where pushing images during an airgap installation just hangs.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:41]: I believe it just appears to hang. Adding my understanding below. Correct if wrong. 

```suggestion
* Fixed an issue where pushing images during an airgap installation hangs when the progress bar is nearly complete. Attempting to reload the page will either fix the problem, or prompt another upload that will work the second time around. 
```

File: content/release-notes/1.17.0.md [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed a bug where automatic update checks result in duplicate releases.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:42]:
```suggestion
* Fixed a bug where automatic update checks result in duplicate releases on the Version History page.
```

File: content/release-notes/1.17.0.md [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+* Fixed a bug where re-uploading an airgap bundle does not work if previous upload fails.
```
Comment (austinchambers) [inline content/release-notes/1.17.0.md:43]: This always happens? When the user is in this situation, how can they fix it?

File: content/release-notes/1.17.0.md (hunk 7 above)
Comment (salahalsaleh) [reply content/release-notes/1.17.0.md:18]: It was force restarting pods for customers, so we decided to get rid of them for the time being and come with a better solution in the future. Also, these annotations are not used at all yet.

File: content/release-notes/1.17.0.md (hunk 9 above)
Comment (salahalsaleh) [reply content/release-notes/1.17.0.md:20]: It was not working and it served no significant purpose.

File: content/release-notes/1.17.0.md (hunk 10 above)
Comment (salahalsaleh) [reply content/release-notes/1.17.0.md:22]: Hmmm, you're right, we should probably move this to the "Fixes" section.
This was not working for:
1- Automated installs using the CLI.
2- For apps with no preflight checks.
3- If "Continue" is not clicked in the preflights page.

File: content/release-notes/1.17.0.md (hunk 15 above)
Comment (salahalsaleh) [reply content/release-notes/1.17.0.md:30]: `kots upstream upgrade`

File: content/release-notes/1.17.0.md (hunk 16 above)
Comment (salahalsaleh) [reply content/release-notes/1.17.0.md:32]: This always happens. And yes, by automated installs i mean using `kots install` and providing a license and config values, which should bypass the config and preflights screen, in which case, the api was not triggering the preflights.

File: content/release-notes/1.17.0.md (hunk 26 above)
Comment (salahalsaleh) [reply content/release-notes/1.17.0.md:43]: Hmmmm, i guess not, it only happens if the previous upload failed while pushing the app archive to s3/docker.

File: content/release-notes/1.17.0.md (hunk 7 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:19]: I'd then put this under the bug section, and say: 
```
Fixed an issue where pods were force-restarted because of KOTS metadata in application manifests. Fixed by ensuring that the system no longer adds KOTS metadata to these manifests. 
```

File: content/release-notes/1.17.0.md (hunk 9 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:21]: Ok

File: content/release-notes/1.17.0.md (hunk 10 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:22]: Sounds like it should be in the "Fixes" section. 

Still don't quite understand what happened. But if you pair that up with the 3 items above, that sounds good to me.

File: content/release-notes/1.17.0.md (hunk 15 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:31]: Then I'd add this release note right below the other `upstream` release note.
```suggestion
* Fixed a bug in [upstream](https://kots.io/kots-cli/upstream/) KOTS CLI where checking for updates shows `There are no application updates available` even when updates are actually available.
```

File: content/release-notes/1.17.0.md (hunk 7 above)
Comment (marccampbell) [reply content/release-notes/1.17.0.md:19]: I think this should stay as a change. Removing this could possibly have adverse effects on an app if they were using undocumented behavior. Specifically this was removed ONLY for gitops enabled clusters (which should be called out)

File: content/release-notes/1.17.0.md (hunk 11 above)
Comment (marccampbell) [reply content/release-notes/1.17.0.md:23]: @austinchambers but kots can be used as a library, and i'd rather include developer release notes than not include them. This is important for someone who might be using kots in a way that we don't document specifically.

File: content/release-notes/1.17.0.md (hunk 18 above)
Comment (marccampbell) [reply content/release-notes/1.17.0.md:36]: This changes the security requirements for KOTS. This should be reworded and kept.

How about:

```Removed the CAP_CHOWN capability from the minio docker file. When deploying to existing clusters, this SELinux capability will no longer be required.```

File: content/release-notes/1.17.0.md (hunk 7 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:20]: In that case, I'd advocate it being both a change and a bug. I'd also call out GitOps enabled clusters. 

```suggestion
* Removed the KOTS metadata that was added to all application manifests. For GitOps enabled clusters, this metadata could result in situations where pods were incorrectly force-restarted. 
```

File: content/release-notes/1.17.0.md (hunk 11 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:24]: I suppose developer release notes are possibly best placed in the repository, and not kots.io. It confuses our audiences. So I still think it's best removed. 

Also curious about how you use KOTS as a library?

File: content/release-notes/1.17.0.md (hunk 18 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:36]: Sounds good to me: 
```suggestion
* Removed the CAP_CHOWN capability from the minio docker file. When deploying to existing clusters, this SELinux capability will no longer be required.
```

File: content/release-notes/1.17.0.md (hunk 7 above)
Comment (marccampbell) [reply content/release-notes/1.17.0.md:20]: This is not correct.  We only added it to GitOps clusters.

File: content/release-notes/1.17.0.md (hunk 11 above)
Comment (marccampbell) [reply content/release-notes/1.17.0.md:24]: It's a go library (actually kotsadm uses it as a go library, not an executable).

Let's reword this to:

* Update kubernetes dependencies to 1.18

File: content/release-notes/1.17.0.md (hunk 16 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:29]:
```suggestion
* Fixed a bug where preflight checks fail to trigger for UI-based airgap installations, or for automated installations where license and configuration has been provided to `kots install` via the CLI. 
```

File: content/release-notes/1.17.0.md (hunk 11 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:24]: Rewording accordingly
```suggestion
* Update to import k8s 1.18 libraries.
```

Comment (marccampbell) [reply content/release-notes/1.17.0.md:24]: I think we broke the github suggested change here.

File: content/release-notes/1.17.0.md (hunk 7 above)
Comment (austinchambers) [reply content/release-notes/1.17.0.md:20]: Ok, tried to update accordingly.

File: content/release-notes/1.17.0.md [hunk 27]
```diff
@@ -0,0 +1,46 @@
...
+
+{{<changes>}}
+* Velero add-on in kURL has been updated to match the official Velero configuration.  This is a breaking change that requires the velero namespace to be deleted before performing the next update with kURL.
+* The most recent support bundles on the Admin Console troubleshoot page are now shown first. 
+* Helpful warnings now appear on the Version History page if the user attempts to deploy a version of the application with invalid YAML. 
+* Removed the KOTS metadata that was added to all application manifests for GitOps enabled clusters. This metadata could result in situations where pods were incorrectly force-restarted. 
+* Improved the granularity of the airgap upload progress indicator.
+* Removed the "Re-analyze" button from the troubleshoot support bundle analysis page.
+* Auto deploy the app's first version when installing the app and no preflight checks are available or preflight checks pass.
+* Update to import k8s 1.18 libraries.
```
Comment (marccampbell) [inline content/release-notes/1.17.0.md:24]: Update kubernetes dependencies to 1.18
